From fca16c8b4612edfa63d7882379897a9907dde738 Mon Sep 17 00:00:00 2001
From: Miroslav Rezanina <mrezanin@redhat.com>
Date: Wed, 10 Jul 2024 02:25:51 -0400
Subject: Remove upstream machine types for aarch64, s390x and x86_64
 architectures
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

We will replace upstream machine types on supported architectures with RHEL
machine types.

Signed-off-by: Daniel P. Berrangé <berrange@redhat.com>
Signed-off-by: Miroslav Rezanina <mrezanin@redhat.com>

Rebase notes (9.1.0):
- Split off commits adding RHEL machine types
---
 hw/arm/virt.c              | 2 ++
 hw/i386/pc_piix.c          | 2 ++
 hw/i386/pc_q35.c           | 2 ++
 hw/s390x/s390-virtio-ccw.c | 3 +++
 4 files changed, 9 insertions(+)

diff --git a/hw/arm/virt.c b/hw/arm/virt.c
index 1800981317..e6e98fef1c 100644
--- a/hw/arm/virt.c
+++ b/hw/arm/virt.c
@@ -3459,6 +3459,7 @@ static void machvirt_machine_init(void)
 }
 type_init(machvirt_machine_init);
 
+#if 0 /* Disabled for Red Hat Enterprise Linux */
 static void virt_machine_10_1_options(MachineClass *mc)
 {
 }
@@ -3634,3 +3635,4 @@ static void virt_machine_4_1_options(MachineClass *mc)
     mc->auto_enable_numa_with_memhp = false;
 }
 DEFINE_VIRT_MACHINE(4, 1)
+#endif /* disabled for RHEL */
diff --git a/hw/i386/pc_piix.c b/hw/i386/pc_piix.c
index c03324281b..acf010e20f 100644
--- a/hw/i386/pc_piix.c
+++ b/hw/i386/pc_piix.c
@@ -475,6 +475,7 @@ static void pc_i440fx_init(MachineState *machine)
 #define DEFINE_I440FX_MACHINE_AS_LATEST(major, minor) \
     DEFINE_PC_VER_MACHINE(pc_i440fx, "pc-i440fx", pc_i440fx_init, true, "pc", major, minor);
 
+#if 0 /* Disabled for Red Hat Enterprise Linux */
 static void pc_i440fx_machine_options(MachineClass *m)
 {
     PCMachineClass *pcmc = PC_MACHINE_CLASS(m);
@@ -802,6 +803,7 @@ static void pc_i440fx_machine_2_6_options(MachineClass *m)
 }
 
 DEFINE_I440FX_MACHINE(2, 6);
+#endif /* Disabled for Red Hat Enterprise Linux */
 
 #ifdef CONFIG_ISAPC
 static void isapc_machine_options(MachineClass *m)
diff --git a/hw/i386/pc_q35.c b/hw/i386/pc_q35.c
index b309b2b378..2203ffd67e 100644
--- a/hw/i386/pc_q35.c
+++ b/hw/i386/pc_q35.c
@@ -374,6 +374,7 @@ static void pc_q35_machine_options(MachineClass *m)
                      pc_q35_compat_defaults, pc_q35_compat_defaults_len);
 }
 
+#if 0 /* Disabled for Red Hat Enterprise Linux */
 static void pc_q35_machine_10_1_options(MachineClass *m)
 {
     pc_q35_machine_options(m);
@@ -685,3 +686,4 @@ static void pc_q35_machine_2_6_options(MachineClass *m)
 }
 
 DEFINE_Q35_MACHINE(2, 6);
+#endif /* Disabled for Red Hat Enterprise Linux */
diff --git a/hw/s390x/s390-virtio-ccw.c b/hw/s390x/s390-virtio-ccw.c
index a79bd13275..2fca2bcf4d 100644
--- a/hw/s390x/s390-virtio-ccw.c
+++ b/hw/s390x/s390-virtio-ccw.c
@@ -911,6 +911,7 @@ static const TypeInfo ccw_machine_info = {
     DEFINE_CCW_MACHINE_IMPL(false, major, minor)
 
 
+#if 0 /* Disabled for Red Hat Enterprise Linux */
 static void ccw_machine_10_1_instance_options(MachineState *machine)
 {
 }
@@ -1167,6 +1168,8 @@ static void ccw_machine_4_2_class_options(MachineClass *mc)
 }
 DEFINE_CCW_MACHINE(4, 2);
 
+#endif /* disabled for RHEL */
+
 static void ccw_machine_register_types(void)
 {
     type_register_static(&ccw_machine_info);
-- 
2.39.3

